### ISSUE TITLE
CPython - IronPython Differences

### ISSUE THREAD

**Issue by DaveMalek** [eirannejad/pyRevit#777]
Hi All,

I am new to programming and I am trying to create an extension using pyrevit. 
Well, I need to use numpy for a part of this extension so I am using new CPython engine.
The first issue that I have encountered is that I used to define points like this (in IronPython):
point = XYZ(1,1,2)

but now (using CPython) when I am using the exact same thing as above and print the point, it always give me (0,0,0).

The codes are pretty simple, you can see them below:

#######################
in IronPython:

from Autodesk.Revit.DB import*
print(XYZ(0,1,0))

########################
in CPython:

#! python3
from Autodesk.Revit.DB import*
print(XYZ(0,1,0))  # this gives me (0,0,0) !

Any help would be much appreciated.
Thanks,
Dave

**Comment by vengelhard** [comment]
Hi, 

I experienced the same when I was trying to create an instance of the TopographySurface class. I'd like to use pandas for fast data operations on an external points file (user input). Therefore it would be great if the XYZ class would also work correctly with CPython.

Is there any news about that issue? I'm doing my master thesis in Revit automation relying on pyRevit. So I would be very happy to know about how to solve this or if there is any work-around.

Thank you so much, 
Verena

**Comment by eirannejad** [comment]
@vengelhard I tested your script in ironpython 277 and python 372 engines and I get the exact same results

![Screen Shot 2020-03-16 at 09 10 55](https://user-images.githubusercontent.com/8197916/76777961-429afc80-6766-11ea-9f47-03c681f8b9f3.png)
![Screen Shot 2020-03-16 at 09 11 47](https://user-images.githubusercontent.com/8197916/76777964-43339300-6766-11ea-896b-900ba77d5561.png)

Not sure what the issue is here

**Comment by vengelhard** [comment]
@eirannejad Thank you for your reply. The script I posted is already meant to be the solution. So... I'm happy that it works also for you. In section Update I described what I did wrong and how I resolved this. Before, I didn't manually create a .NET list in CPython which IronPython does automatically. Also I passed one coordinate tuple to XYZ class instead of three coordinate values (just wondering about why I was thinking that this would work). Almost sure that DaveMaleks result was caused by something else. Thanks again Ehsan!
